pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -334,16 +335,28 @@ def _load_config(self, config):
         return config
 
     def setup_distributed(self):
-        super().setup_distributed()
+        reset_seed()
+
+        # determine which process we are and world size
+        self.set_world_ranks()
+
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+
+        self.init_deepspeed_distributed()
+
+        # set the ranks and devices
+        self.dist.rank = self.global_rank
+        self.dist.device = self.root_device
         if not self._config_initialized:
             self._format_config()
             self._config_initialized = True
 
-    def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Optional[int] = None) -> None:
+    def init_deepspeed_distributed(self) -> None:
         if platform.system() != "Windows":
             # do not set env variables on windows, allow deepspeed to control setup
-            global_rank = global_rank if global_rank is not None else self.cluster_environment.global_rank()
-            world_size = world_size if world_size is not None else self.cluster_environment.world_size()
+            global_rank = self.global_rank if self.global_rank is not None else self.cluster_environment.global_rank()
+            world_size = self.world_size if self.world_size is not None else self.cluster_environment.world_size()
```

Comment:
noob q: why can't we just use the cluster environment directly?

probably for @awaelchli

Reply:
general answer: for plugins (especially custom ones) that don't define a cluster env you would need a none check here which is what the properties do. One could also define a default cluster env that returns rank 0 and world size 1, we haven't explored that. 

for deepspeed specifically, you can probably assume there always is an environment so None is not allowed.